spec/models/poly_metadatum_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PolyMetadatum, type: :model do
+  subject(:test_metadatum) { build(:poly_metadatum) }
+
+  # Tests for validations
+  describe 'validations' do
+    it 'requires a key' do
+      expect(test_metadatum).to validate_presence_of(:key)
+    end
+
+    it 'requires a value' do
+      expect(test_metadatum).to validate_presence_of(:value)
+    end
+
+    it 'key must be unique' do
+      expect(test_metadatum).to validate_uniqueness_of(:key).scoped_to(:metadatable_id).case_insensitive
+    end
+
+    context 'when creating a new poly_metadatum' do
+      it 'is valid with valid attributes' do
+        expect(test_metadatum).to be_valid
+      end
+
+      it 'is invalid without a key' do
+        test_metadatum.key = nil
+        expect(test_metadatum).not_to be_valid
+        expect(test_metadatum.errors.full_messages).to include('Key can\'t be blank')
+      end
```

Comment:
Are these tests not duplicating what the `validate_presence_of` tests do above?

Reply:
Yes you are right I will refactor a bit